### PR TITLE
Added character entity support

### PR DESCRIPTION
**Pull request by nihal111**
Linked with https://github.com/Terasology/WildAnimals/pull/9 and https://github.com/MovingBlocks/Terasology/pull/2929

This adds Worldly Tooltip integration for character entities, specifically WildAnimals.

Tried adding a box around the ui, which doesn't seem to resize to `"use-content-height"`. Encountered some bugs with NUI and NUIEditor and reported them as engine issues. Will possibly fix the UI later in a follow up PR.

Looks like this:
screenshot from 2017-05-03 22-51-44
screenshot from 2017-05-03 23-37-33

### CODE REVIEW

**Comment by nihal111** [comment]
@Josharias Can you please review this?

File: src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java [hunk 1]
```diff
@@ -111,8 +122,7 @@ public String get() {
                         @Override
                         public List<TooltipLine> get() {
                             if (cameraTargetSystem.isTargetAvailable()) {
-                                EntityRef targetEntity = blockEntityRegistry.getEntityAt(cameraTargetSystem.getTargetBlockPosition());
-
+                                EntityRef targetEntity = cameraTargetSystem.getTarget();
```
Comment (jzacharias-associa) [inline src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java:125]: This works ok for blocks too, right?  If so,  I wonder why I chose to use the block entity registry... Good job finding a more generic solution.

File: src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java [hunk 2]
```diff
@@ -128,18 +138,38 @@ public String get() {
...
         icon = find("icon", ItemIcon.class);
         if (icon != null) {
+            icon.bindIcon(new ReadOnlyBinding<TextureRegion>() {
+                @Override
+                public TextureRegion get() {
+                    if (cameraTargetSystem.isTargetAvailable()) {
+                        if (!cameraTargetSystem.isBlock()) {
+                            EntityRef targetEntity = cameraTargetSystem.getTarget();
+                            GetTooltipIconEvent getTooltipIconEvent = new GetTooltipIconEvent();
+                            targetEntity.send(getTooltipIconEvent);
```
Comment (jzacharias-associa) [inline src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java:148]: This is a much better pattern than the hardcoding I did for block entities.  I suppose the natural evolution would use a ```GetTooltipMeshEvent``` to get the mesh for blocks (not now,  just thoughts for the future)

**Review by jzacharias-associa — APPROVED** [review]
I also had troubles with the box.  I am glad you found out why the autosizing was not working correctly.  I didnt dive far enough into the root cause.

If you dont mind just doing a quick verification that the block tooltips still work before merging,  that would be great.

One more thing,  could you bump the minor version of the module to 1.1 to mark the addition of this functionality?

One last thing, those screenshots are gold.  I would be quite happy if you added them to the readme as well.

File: src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java (hunk 1 above)
Comment (nihal111) [reply src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java:125]: Yes it works for blocks too. :slightly_smiling_face:

File: src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java (hunk 2 above)
Comment (nihal111) [reply src/main/java/org/terasology/worldlyTooltip/ui/WorldlyTooltip.java:148]: Yes probably can have that in the future.
FYI, these events are defined in the WorldlyTooltip module at present. The module needs to be added as an optional dependency (new feature- https://github.com/MovingBlocks/Terasology/pull/2767) for these events to be handled.

**Comment by nihal111** [comment]
Of course the block tooltip still works. :smile: 


Added screenshots in readme and bumped version.
Merging now. Thanks for the review @jzacharias-associa @Josharias

**Pull request merged**